nemo/collections/nlp/parts/nlp_overrides.py

```diff
@@ -410,6 +412,9 @@ def save_checkpoint(
             if self.is_global_zero or app_state.data_parallel_rank == 0:
                 self.checkpoint_io.save_checkpoint(checkpoint, filepath, storage_options=storage_options)
 
+        if self.save_last_n_optim_states >= 0 and '-last' in filepath:
```

Comment:
This creates a big entanglement in the code.

`-last` is very specific to ModelCheckpoint callbacks, the strategy shouldn't know about stuff like this and it shouldn't happen during `save_checkpoint` call